syntax = "proto2";

package config;

message CalendarConfig {
	optional int32 year = 1;
	optional int32 month = 2;

	optional double margin_top = 25 [default = 10];

	optional double cell_width = 3;
	optional double cell_height = 4;
	optional double cell_margin = 5 [default = 10];
	optional double line_width = 6 [default = 1];

	optional OutputType output_type = 7 [default = SVG];
	optional PaperType paper_type = 8 [default = US_LETTER];
	optional Language language = 9 [default = ENGLISH];

	optional CoverageType coverage_type = 22 [default = WHOLE_BIBLE];
	optional DurationType duration_type = 23 [default = ONE_YEAR];
	repeated DayOfTheWeek days_to_rest = 24;

	optional string output_file_name = 11 [default = "output"];

	optional string default_font_family = 12;

	optional string day_number_font_family = 13;
	optional double day_number_font_size = 14 [default = 30];

	optional string day_plan_font_family = 15;
	optional double day_plan_font_size = 16 [default = 20];

	optional string month_label_font_family = 17;
	optional double month_label_font_size = 18;
	optional bool month_label_uppercase = 19 [default = false];

	optional string wday_label_font_family = 20;
	optional double wday_label_font_size = 21 [default = 16];
}

enum OutputType {
	SVG = 0;
	PDF = 1;
	PNG = 2;
}

enum PaperType {
	US_LETTER = 0;
	A4 = 1;
}

enum Language {
	ENGLISH = 0;
	KOREAN = 1;
}

enum CoverageType {
	NEW_TESTAMENT = 0;
	OLD_TESTAMENT = 1;
	NEW_TESTAMENT_AND_PSALMS = 2;
	WHOLE_BIBLE = 3;
	WHOLE_BIBLE_NEW_TESTAMENT_FIRST = 4;
	WHOLE_BIBLE_IN_PARALLEL = 5;
}

enum DurationType {
	ONE_YEAR = 0;
	TWO_YEARS_FIRST_YEAR = 1;
	TWO_YEARS_SECOND_YEAR = 2;
}

enum DayOfTheWeek {
	SUN = 0;
	MON = 1;
	TUES = 2;
	WED = 3;
	THURS = 4;
	FRI = 5;
	SAT = 6;
}
